lib/ansible/modules/web_infrastructure/jenkins_job.py

```diff
@@ -341,7 +341,8 @@ def test_dependencies(module):
 
 
 def job_config_to_string(xml_str):
-    return ET.tostring(ET.fromstring(xml_str))
+    xml_element_tree = ET.fromstring(ensure_binary(xml_str))
+    return ensure_str(ET.tostring(xml_element_tree))
```

Comment:
This added ensure_str looks like it should be to_text/ensure_text instead.  The jenkins API is called on the output of this here: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/web_infrastructure/jenkins_job.py#L284  The jenkins API calls encode on that here: https://opendev.org/jjb/python-jenkins/src/branch/master/jenkins/__init__.py#L1263   So, on python2, if this is a byte string with non-ascii characters present, it will traceback.  It needs to be a text string instead.

Reply:
The change on line #L293 of the current PR should be the one you linked, which is also altered now.
I should verify in my test set up that special chars are passed correctly, thought they were

Reply:
Local set up with python2.7 works fine with utf-8 content in the xml file. It does give an error when jenkins_job is called with utf-8 characters in params such as `name`. This however also happens with the latest code from #54958